### PR TITLE
Add GHA workflow to detect link rot

### PR DESCRIPTION
**Pull request by IndrajeetPatil**
Inspired by this comment (https://github.com/r-lib/lintr/pull/1690#discussion_r993816614)

### CODE REVIEW

File: .github/workflows/check-link-rot.yaml [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+            any::rcmdcheck
+            any::urlchecker
+
+      - name: Run URL checker
+        run: |
+          options(crayon.enabled = TRUE)
+          rotten_links <- urlchecker::url_check(progress = FALSE)
+          print(rotten_links)
+          if (length(rotten_links$URL) > 0L) {
+            stop("Some URLs are outdated and need to be updated.", call. = FALSE)
```
Comment (AshesITR) [inline .github/workflows/check-link-rot.yaml:39]: Will the concrete links and locations be visible?

File: .github/workflows/check-link-rot.yaml [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+jobs:
+  check-link-rot:
+    runs-on: ubuntu-latest
+    env:
+      GITHUB_PAT: ${{ secrets.GITHUB_TOKEN }}
+      R_KEEP_PKG_SOURCE: yes
+    steps:
+      - uses: actions/checkout@v3
+
+      - uses: r-lib/actions/setup-pandoc@v2
```
Comment (AshesITR) [inline .github/workflows/check-link-rot.yaml:18]: Is pandoc necessary here?

File: .github/workflows/check-link-rot.yaml (hunk 1 above)
Comment (IndrajeetPatil) [reply .github/workflows/check-link-rot.yaml:39]: > Will the concrete links and locations be visible?

Yes, for a failed workflow example, see [this](https://github.com/IndrajeetPatil/minimalReprex/actions/runs/3434717367/jobs/5726296581) run. 

Also, for an example of a successful run, see [this](https://github.com/IndrajeetPatil/statsExpressions/actions/runs/3434847347/jobs/5726551334) one.

File: .github/workflows/check-link-rot.yaml (hunk 2 above)
Comment (IndrajeetPatil) [reply .github/workflows/check-link-rot.yaml:18]: Yes, definitely. I had tried without (see commit history in this [experimental PR](https://github.com/IndrajeetPatil/minimalReprex/pull/8)) and it fails.

File: .github/workflows/check-link-rot.yaml (hunk 1 above)
Comment (AshesITR) [reply .github/workflows/check-link-rot.yaml:39]: Cool, thanks!

**Review by AshesITR — APPROVED** [review]

**Pull request merged**